### ISSUE TITLE
Paging for search

### ISSUE THREAD

**Issue by DenaZH** [LMFDB/lmfdb#3334]
When a list of  examples show up with a search, it would be nice to have a first page & last page option in addition to previous and next buttons.

**Comment by roed314** [comment]
First button is pretty easy.

The last button is more complicated, though doable if we decide we want it.  It will require computing the total number of results (which can take a little while in some cases).  Then a large offset takes a longer time for the SQL query to execute.  To get an idea of what the performance would look like, compare
 * http://www.lmfdb.org/NumberField/?class_number=1
 * http://www.lmfdb.org/NumberField/?start=10030000&class_number=1
The second takes far longer to load, and is pretty similar to what the load time would be for a last button on such a search.

Given the performance issue, I'm not sure we should implement either button, but I'm willing to be convinced.

**Comment by DenaZH** [comment]
I see. The second link doesn't even open for me. Maybe just a first button for now would be good enough. Thanks!

**Comment by alexjbest** [comment]
I like this idea. Currently I end up doing the following often : clicking the "of more than 1000" button to see the total number of results, rounding it down to the nearest 50 moving to the second page so that the url has the start=50 in it and then changing 50 to the number I just worked out. Seeing as people are probably doing this anyway (if they worked out how) I dont see why we should make people jump through hoops just because its not performant. (Unless you are worried about web scrapers and crawlers pressing the button regularly, but they aren't going inputing complex queries anyway probably).

**Comment by roed314** [comment]
Can't one just reverse the search order instead of implement a last page button?

**Comment by jenpaulhus** [comment]
True.  Although I'll note that it may not be immediately obvious to newer users that you can double click sort order to change the order.   Could we add a Last button after the Next that does exactly the reverse search order?
